miden-tx/src/host/tx_authenticator.rs

```diff
@@ -39,15 +42,54 @@ pub trait TransactionAuthenticator {
     ) -> Result<Vec<Felt>, AuthenticationError>;
 }
 
-// BASIC AUTHENTICATOR
+// AUTH SECRET KEY
 // ================================================================================================
 
 /// Types of secret keys used for signing messages
 #[derive(Clone, Debug)]
+#[repr(u8)]
 pub enum AuthSecretKey {
-    RpoFalcon512(rpo_falcon512::SecretKey),
+    RpoFalcon512(rpo_falcon512::SecretKey) = 0,
+}
+
+impl AuthSecretKey {
+    /// Identifier for the type of authentication key
+    pub fn key_id(&self) -> u8 {
+        match self {
+            AuthSecretKey::RpoFalcon512(_) => 0u8,
+        }
+    }
+}
+
+impl Serializable for AuthSecretKey {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        let mut bytes = vec![self.key_id()];
+        match self {
+            AuthSecretKey::RpoFalcon512(key_pair) => {
+                bytes.append(&mut key_pair.to_bytes());
+                target.write_bytes(&bytes);
+            },
+        }
```

Comment:
Do we need to allocate the vector here? Maybe something like this will work:
```Rust
self.key_id().write_into(target);
match self {
    AuthSecretKey::RpoFalcon512(secret_key) => {
        secret_key.write_into(target);
    },
}
```